shared/src/test/scala/scala/xml/XMLTest.scala

```diff
@@ -369,6 +369,14 @@ Ours is the portal of hope, come as you are."
     </wsdl:definitions>""", wsdlTemplate4("service4", () => "target4") toString)
   }
 
+  @UnitTest
+  def t547: Unit = {
+    // ambiguous toString problem from #547
+
+    val atom: scala.xml.Atom[Unit] = new scala.xml.Atom(())
+    assertEquals("()", atom.toString)
+  }
+
```

Comment:
Scalajs complains for this test:

```
> xmlJS/testOnly scala.xml.XMLTest
[error] Test scala.xml.XMLTest.t547 failed: expected:<[(])> but was:<[undefine])>, took 0.001000192 sec
[error] Failed: Total 32, Failed 1, Errors 0, Passed 31
```

The string representation of `Unit` is different in Scalajs?

Reply:
Yes, in Scala.js, `().toString == "undefined"`, because `()` to the JavaScript value `undefined`, which is what a JavaScript function returns when it doesn't explicit `return` something else.